src/pages/receipts/packagesInTransit.js

```diff
@@ -88,15 +88,26 @@ const packagesInTransitTemplate = async (username, auth, route) => {
 };
 
 /**
- * Returns an array of shipped box items but not yet received and sorts the box items by most recent date
- * @param {array} boxes - array of all boxes from getAllBoxes(`bptl`) function
- * @returns sorted array of boxes by most recent date
+ * filteredBoxesBySubmitShipmentTimeAndNotReceived(): handle boxes with a received shipment flag with a default false value.
+ * sortBoxesBySubmitShipmentTime(): sort method use direct comparison operator over the localeCompare for more speed.
  */
 export const getRecentBoxesShippedBySiteNotReceived = (boxes) => {
-  // boxes are from searchBoxes endpoint
+    // boxes are from searchBoxes endpoint
     if (boxes.length === 0) return [];
-    const filteredBoxesBySubmitShipmentTimeAndNotReceived = boxes.filter(item => item[fieldToConceptIdMapping["shippingShipDate"]] && !item[fieldToConceptIdMapping["siteShipmentDateReceived"]])
-    const sortBoxesBySubmitShipmentTime = filteredBoxesBySubmitShipmentTimeAndNotReceived.sort((a,b) => b[fieldToConceptIdMapping["shippingShipDate"]].localeCompare(a[fieldToConceptIdMapping["shippingShipDate"]]))
+
+    const filteredBoxesBySubmitShipmentTimeAndNotReceived = boxes.filter(item => {
+        const hasShippingShipDate = item[fieldToConceptIdMapping.shippingShipDate];
```

Comment:
I think we're okay here because the item object's property key is a number with a value that's a string ->
object[number]: 'string' 
item[key]: 'value'
item[656548982]: '2022-10-20T00:19:27.546Z'

@Gbarra9 could you weigh in on this?

Reply:
From Joe's above comment, I think we should be good here. While `[fieldToConceptIdMapping.shippingShipDate]` is referencing a concept Id with number data type value, the bracket notation should automatically convert the number type to a string type. I think a change is not needed unless this is bad practice.

Reply:
Additional note per Gene's suggestion -> updating 'item' to 'boxObj' - it's more descriptive and goes well with naming conventions in other parts of the app.